### ISSUE TITLE
Optional chaining reports unorthodox function invocation error in specific case

### ISSUE THREAD

**Issue by ashwinbaliga** [jshint/jshint#3551]
Using the latest JSHint 2.13.0 that has optional chaining support (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), I see that a specific line wrongly reports an "unorthodox function invocation" error:

```
/* jshint esversion:11 */
var someChainingObj = {aNum:10,
                       aFunc: function() { return 500; },
                       anArr: ['val1', 'val2'] };

alert(someChainingObj?.aFunc?.()); // Reports unorthodox function invocation error
alert(someChainingObj.aFunc?.()); // Works fine

alert(someChainingObj?.anArr?.[1]); // Works fine
alert(someChainingObj.anArr?.[1]); // Works fine

alert(someChainingObj?.aNum?.isPrototypeOf); // Works fine
alert(someChainingObj.aNum?.isPrototypeOf); // Works fine

```

The error is only for the above mentioned line for "aFunc". It does not happen for arrays and for objects. I don't see anything wrong in the line that reports the above error.

**Comment by jugglinmike** [comment]
Thanks for the report! The issue is the CallExpression, so we can futher reduce the problem like this:

```js
/* jshint expr: true, esversion: 11 */

Math?.min();
```

Working on a fix, now.

**Comment by jugglinmike** [comment]
Fix under review here: gh-3552

**Comment by jugglinmike** [comment]
[We released the fix in JSHint version 2.13.1.](https://jshint.com/blog/2021-08-10/release-2-13-1/)
